test/integration/backward_compatible/parallel/topic/TopicTest.js

```diff
@@ -0,0 +1,163 @@
+    });
+
+    beforeEach(async function () {
+        topic = await client.getTopic('topic-test');
+    });
+
+    afterEach(async function () {
+        return topic.destroy();
+    });
+
+    it('tests listener', async function () {
+        class Collector {
+            constructor() {
+                this.events = [];
+            }
+
+            onMessage(event) {
+                event.publish_time = Date.now();
+                this.events.push(event);
+            }
+        }
+
+        const collector = new Collector();
+
+        setTimeout(() => {
+            expect(collector.events.length).to.be.equal(1);
+            const event = collector.events[0];
+            expect(event.message).to.be('item-value');
+            expect(event.publish_time).to.be.above(0);
+        }, 5000);
+    });
+
+    it('removes listener', async function() {
+        class Collector {
+            constructor() {
+                this.events = [];
+            }
+
+            onMessage(event) {
+                this.events.push(event);
+            }
+        }
+
+        const collector = new Collector();
+        const reg_id = await topic.addListener(collector.onMessage.bind(collector));
+        await topic.removeListener(reg_id);
+        await topic.publish('item-value');
+
+        setTimeout(() => {
+            expect(collector.events.length).to.be(0);
+        }, 5000);
+    });
+
+    it('tests publish', async function () {
+        let count = 0;
+        const receivedValues = [];
+
+        await topic.addListener((message) => {
+            count++;
+            receivedValues.push(message.messageObject);
+        });
+
+        const message = 'message';
+        await topic.publish(message);
+
+        await new Promise((resolve) => {
+            const interval = setInterval(() => {
+                if (count === 1) {
+                    clearInterval(interval);
+                    expect(receivedValues).to.deep.equal([message]);
+                    resolve();
+                }
+            }, 100);
+        });
+    });
+
+    it('tests publishAll', async function() {
+        let count = 0;
+        const receivedValues = [];
+
+        await topic.addListener((message) => {
+            count++;
+            receivedValues.push(message.messageObject);
+        });
+
+        const messages = ['message 1', 'message 2', 'message 3'];
+        await topic.publishAll(messages);
+
+        await new Promise((resolve, reject) => {
+            const interval = setInterval(() => {
+                if (count === messages.length) {
+                    clearInterval(interval);
+                    expect(receivedValues).to.have.members(messages);
+                    resolve();
+                }
+            }, 100);
+            setTimeout(() => {
+                clearInterval(interval);
+                reject(new Error('Timed out while waiting for messages to be published'));
+            }, 5000);
+        });
```

Comment:
Again we have assertTrueEventually methods for this kind of logic.

Reply:
Okay, I see